core/trino-main/src/test/java/io/trino/sql/planner/TestDomainTranslator.java

```diff
@@ -834,6 +834,16 @@ public void testFromBasicComparisonsWithNaN()
         assertUnsupportedPredicate(not(isDistinctFrom(C_REAL, nanReal)));
     }
 
+    @Test
+    public void testFromCoercionComparisonsWithNaN()
+    {
+        Expression nanDouble = literalEncoder.toExpression(TEST_SESSION, Double.NaN, DOUBLE);
+
+        assertUnsupportedPredicate(equal(cast(C_TINYINT, DOUBLE), nanDouble));
+        assertUnsupportedPredicate(equal(cast(C_SMALLINT, DOUBLE), nanDouble));
+        assertUnsupportedPredicate(equal(cast(C_INTEGER, DOUBLE), nanDouble));
```

Comment:
In this particular case we could also simplify this to false.
Wonder whether this can be generalized... Can `floorValue` failure imply that this is an always-false case?

cc @martint

Reply:
I don't know, @martint WDYT?

Reply:
@martint thoughts?

Reply:
It depends on how floorValue is being used. In the context of doing an inference similar to what we do in UnwrapCastInComparison, yes, that would be a safe assumption -- it means there's no value in one domain that matches any value in the other domain.